components/cam/src/physics/cam/crm_physics.F90

```diff
@@ -738,15 +741,19 @@ subroutine crm_physics_tend(ztodt, state, tend, ptend, pbuf, cam_in, cam_out,
    !------------------------------------------------------------------------------------------------
    ! Set CRM orientation angle
    !------------------------------------------------------------------------------------------------
-
+   crm_angle(:) = 0
```

Comment:
Didn't we agree that we don't want to initialize this so that it causes problems if it's not set?

Reply:
Sorry, I don't follow. What kinds of problem would you expect to catch by not initializing it?

In the static configurations (NS and EW) it gets set to the proper value regardless, so initializing does no harm. In the `SP_ORIENT_RAND` code block prior to this change, it had been using those uninitialized values in the form of `crm_angle(i) = crm_angle(i) + stuff`, so not initializing prior to the first use will give unpredictable results.

Reply:
I'm thinking about an issue where someone tries to implement some scheme for setting the orientation, like making is depend on the large-scale wind as in the early Grabowski paper, but they mess up and don't set all the columns, then if crm_angle is initialized they will get the a different orientation than what they are expecting, whereas I think it's preferable for it to crash in that case.

For the SP_ORIENT_RAND block we should initialize on the first time step, I like that fix, but it seems like ti should be contained within that block.

Reply:
I don't think leaving it uninitialized helps with the issue you described because it won't necessarily crash if crm_angle is uninitialized. I think it will just use whatever is in that memory slot. For example, this simple program compiles and runs just fine for me, and does not give consistent results from run-to-run:
```
program uninitialized_angle_test
  implicit none
  integer, parameter :: dp=kind(0.d0)  ! double precision                                                                                                       
  real(dp) :: angle(50)
  integer :: i

do i = 1, 50
  print *, 'cos(angle(', i, ')) = ', cos(angle(i))
enddo

end program
```

Reply:
good point... what can we initialize it with that will make it crash?